### ISSUE TITLE
Model fails after adding clustering

### ISSUE THREAD

**Issue by inkrement** [TsaiLintung/fastdid#6]
I just added clustervar to a model that worked; after I made the change it did not. The clsutervar variable is a string-vector and the error message is quite cryptic: 

```R
Error in whichNA(dt[[col]]): Unsupported type 'NULL' passed to whichv()
Traceback:

1. fastdid(data = ssdf, timevar = "period", cohortvar = "tp", unitvar = "customer", 
 .     outcomevar = "XXX", result_type = "dynamic", control_option = "never", 
 .     clustervar = "customer", boot = TRUE, control_type = "dr", 
 .     varycovariatesvar = c("s1", "s2", "s3", "s4", "s5", "s6"))
2. validate_did(dt, varnames, p)
3. whichNA(dt[[col]])
```

Any idea what could be the issue? There are no NAs in the vector.

**Comment by TsaiLintung** [comment]
Thank you for reporting this issue. TL;DR: The bug has been patched in 0.9.3.1 . 

The bug happens because fastdid coerces the column names for `timevar`, `cohortvar`, and `unitvar` to internal names. So when `clustervar` coincides with `unitvar`, there is no column with the name `clustervar` after coercion. Therefore, `dt[[clustervvar]]` returns `NULL` and hence the bug. 

Thankfully, fastdid defaults at clustering the unit level anyway, so this is currently fixed by simply ignoring `clustervar` when `clustervar == unitvar`. But this also makes me realize that the name-changing may cause other problems, so I will work on a more systematic solution in the next big patch. 

In all, thank you very much for reporting this bug, and please let me know if there are any problems with using `fastdid`. Cheers.

**Comment by inkrement** [comment]
Perfect, and thank you for the swift reply. I'll close the issue directly.
